airflow/www/static/js/dag/details/graph/utils.ts

```diff
@@ -0,0 +1,116 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Color from 'color';
+import type {
+  Node as ReactFlowNode,
+} from 'reactflow';
+import type { ElkExtendedEdge } from 'elkjs';
+
+import type { CustomNodeProps } from './Node';
+
+export const nodeColor = (
+  { data: { height, width, instance } }: ReactFlowNode<CustomNodeProps>,
+) => {
+  if (!height || !width) return '';
+  if (width > 200 || height > 60) {
+    return '#cccccc50';
+  }
+  if (instance?.state) {
+    return `${Color(stateColors[instance.state]).hex()}90`;
+  }
+  return '#cccccc90';
+};
+
+export const nodeStrokeColor = (
+  { data: { isSelected } }: ReactFlowNode<CustomNodeProps>,
+  colors: any,
+) => (isSelected ? colors.blue[500] : '');
+
+interface BuildEdgesProps {
+  edges?: ElkExtendedEdge[];
+  nodes: ReactFlowNode<CustomNodeProps>[];
+  taskId?: string | null;
+}
+
+export const buildEdges = ({
```

Comment:
This is the edge data itself in a format that works for react-flow. Edge.tsx is the component itself. I can improve the names or add a comment to avoid confusion